spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/OpenTelemetryConfigurations.java

```diff
@@ -0,0 +1,151 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.tracing;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.micrometer.tracing.otel.bridge.OtelBaggageManager;
+import io.micrometer.tracing.otel.bridge.OtelCurrentTraceContext;
+import io.micrometer.tracing.otel.bridge.OtelTracer;
+import io.micrometer.tracing.otel.bridge.OtelTracer.EventPublisher;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import org.springframework.boot.SpringBootVersion;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+/**
+ * Configurations for Open Telemetry. Those are imported by
+ * {@link OpenTelemetryAutoConfiguration}.
+ *
+ * @author Moritz Halbritter
+ */
+class OpenTelemetryConfigurations {
+
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(SdkTracerProvider.class)
+	@EnableConfigurationProperties(TracingProperties.class)
+	static class SdkConfiguration {
+
+		/**
+		 * Default value for application name if {@code spring.application.name} is not
+		 * set.
+		 */
+		private static final String DEFAULT_APPLICATION_NAME = "application";
+
+		@Bean
+		@ConditionalOnMissingBean
+		OpenTelemetry openTelemetry(SdkTracerProvider sdkTracerProvider, ContextPropagators contextPropagators) {
+			return OpenTelemetrySdk.builder().setTracerProvider(sdkTracerProvider).setPropagators(contextPropagators)
+					.build();
+		}
+
+		@Bean
+		@ConditionalOnMissingBean
+		SdkTracerProvider otelSdkTracerProvider(Environment environment, List<SpanProcessor> spanProcessors,
+				Sampler sampler) {
+			String applicationName = environment.getProperty("spring.application.name", DEFAULT_APPLICATION_NAME);
```

Comment:
Similar to my previous comments on `BraveAutoConfiguration`, it would be good not to have two separate places both with the same fallback logic. Perhaps we can move it out somewhere more central. Again, this could maybe be done under a separate issue with broader scope than just tracing.

Reply:
See https://github.com/spring-projects/spring-boot/issues/30650